### PR TITLE
Add drawer to app

### PR DESCRIPTION
**Pull request by djnunez-aot**
Worked on sidebar navigation and added some mobile changes. Still need to clean up code but wanted everyone to take a look.
Github issue #65

### CODE REVIEW

File: met-web/src/components/layout/Drawer/CustomDrawer.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            height: '100%',
+            color: 'white',
+            width: 'auto',
+            justifyContent: 'center',
+            alignItems: 'center',
+        }}
+        role="presentation"
+    >
+        <List>
+            {['Engagements', 'Surveys', 'Calendar', 'Reporting'].map((text, index) => (
```
Comment (jadmsaadaot) [inline met-web/src/components/layout/Drawer/CustomDrawer.tsx:21]: the links here should be customizable and not hard coded. They could pass as props to the drawer component. We can define them in another file where we can change them if needed. Also They should pass as objects { name: "engagement", path: "/engagement",...}

File: met-web/src/components/layout/Drawer/CustomDrawer.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    </Box>
+);
+
+const Drawer = () => {
+    return (
+        <AppBar
+            className="font-BCBold"
+            position="absolute"
+            style={{
+                backgroundColor: '#003366',
```
Comment (jadmsaadaot) [inline met-web/src/components/layout/Drawer/CustomDrawer.tsx:55]: I noticed there was a lot of customization when it came to creating the app bar. Sure we are taking an MUI component, but we are heavily customizing it and it feels we are rebuilding our own app bar. If we absolutely need that, we would do it. But first, we it's better to explore other side bar component solutions out there. The less customization from us the better

File: met-web/src/routes/LandingPage/LandingPageElements.tsx [hunk 3]
```diff
@@ -3,5 +3,7 @@ import { Row, Column } from '../../components/common';
 
 export const SearchContainer = styled(Row)`
```
Comment (jadmsaadaot) [inline met-web/src/routes/LandingPage/LandingPageElements.tsx:4]: we don't need to build our own search container. We can use material UI's search bar. Check https://mui.com/material-ui/react-text-field/ the Customization section

**Comment by jadmsaadaot** [comment]
This solution could work. It's better to explore ready-made solutions that require much less customization from us. Let's try to avoid heavily customizing or building these components from scratch, unless we absolutely have to.

**Review by jadmsaadaot — CHANGES_REQUESTED** [review]
This solution could work. It's better to explore ready-made solutions that require much less customization from us. Let's try to avoid heavily customizing or building these components from scratch, unless we absolutely have to.

File: met-web/src/components/layout/Drawer/CustomDrawer.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import AppBar from '@mui/material/AppBar';
+import { Grid, ListItemButton, List, ListItem, ListItemText, Box } from '@mui/material';
+import { useAppSelector } from '../../../hooks';
+import UserService from '../../../services/UserServices';
+import { useNavigate } from 'react-router-dom';
+
+const Routes = [
+    { name: 'Engagements', path: '/' },
```
Comment (saravanpa-aot) [inline met-web/src/components/layout/Drawer/CustomDrawer.tsx:9]: may be we need a better pattern to keep all the routes togother at some place so its easier to find.. we can discuss that later..not as a part of this PR anyway

File: met-web/src/components/layout/Drawer/CustomDrawer.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+];
+
+const list = (navigate) => (
+    <Box role="presentation">
+        <List>
+            {Routes.map((route, index) => (
+                <ListItem key={route.name}>
+                    <ListItemButton onClick={() => navigate(route.path)}>
+                        <ListItemText
+                            primaryTypographyProps={{ variant: 'h6', sx: { fontWeight: 'bold' } }}
```
Comment (saravanpa-aot) [inline met-web/src/components/layout/Drawer/CustomDrawer.tsx:22]: can this styles be a class ? or is this best way? like fontWeigh bold can it be avoided?

File: met-web/src/components/layout/SideNav/SideNav.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        </Box>
+    );
+};
+
+const SideNav = ({ open }) => {
+    const navigate = useNavigate();
+    let width = screen.width;
+
+    useEffect(() => {
+        console.log('OPEN CHANGED::' + JSON.stringify(open));
```
Comment (jadmsaadaot) [inline met-web/src/components/layout/SideNav/SideNav.tsx:41]: is this debug needed?

File: met-web/src/routes/AuthenticatedRoutes.tsx [hunk 7]
```diff
@@ -8,17 +8,18 @@ import UserService from '../services/UserServices';
...
     let adminRole = UserService.hasAdminRole();
-
+    let width = screen.width;
     //ToDO: remove when roles are defined in keycloak
     adminRole = true;
 
     return (
         <ThemeProvider theme={BaseTheme}>
-            <Drawer />
+            {width > 1350 ? <SideNav /> : <></>}
```
Comment (jadmsaadaot) [inline met-web/src/routes/AuthenticatedRoutes.tsx:21]: I noticed in the sideNav there was a conditional to view it according the the width > 1350 condition. But here we aren't viewing it at all under 1350. Do we want to hide under 1350 for now?

File: met-web/src/components/layout/Header/HeaderElements.tsx [hunk 8]
```diff
@@ -9,6 +9,13 @@ export const sxLight = {
     boxShadow: '0px 5px 9px #29',
 };
 
+export const headerSx = {
```
Comment (jadmsaadaot) [inline met-web/src/components/layout/Header/HeaderElements.tsx:12]: was this used?

File: met-web/src/components/layout/Header/index.tsx [hunk 9]
```diff
@@ -1,66 +1,74 @@
...
-} from './HeaderElements';
+import { LogoContainer, LogoutContainer, TitleContainer, AuthButton, HeaderText } from './HeaderElements';
 import sx from 'mui-sx';
+import SideNav from '../SideNav/SideNav';
 
 const Header = () => {
     //states
     const isLoggedIn = useAppSelector((state) => state.user.authentication.authenticated);
+    const [open, setOpen] = useState(false);
+    const [width, setWidth] = useState(window.innerWidth);
```
Comment (jadmsaadaot) [inline met-web/src/components/layout/Header/index.tsx:15]: do we still need this?

File: met-web/src/routes/LandingPage/LandingPage.tsx [hunk 10]
```diff
@@ -2,16 +2,16 @@ import React from 'react';
...
 import { Link } from 'react-router-dom';
+import TextField from '@mui/material/TextField';
 
 const LandingPage = () => {
     return (
         <Container sx={{ paddingTop: '5em' }}>
             <Grid direction="row" justifyContent="center" alignItems="center" container spacing={2}>
                 <Grid item xs={9}>
-                    <SearchContainer></SearchContainer>
+                    <TextField sx={{ width: '30%', fontSize: '40px' }} placeholder={'Search Engagements by name...'} />
```
Comment (jadmsaadaot) [inline met-web/src/routes/LandingPage/LandingPage.tsx:14]: Let's avoid tapping into width and height props of components. In the case above we can use fullWidth prop and control the width by the xs prop in the Grid item above. This way there's a conflict between the column span of the grid and the textField width

File: met-web/src/routes/LandingPage/LandingPageElements.tsx (hunk 3 above)
Comment (jadmsaadaot) [inline met-web/src/routes/LandingPage/LandingPageElements.tsx:4]: are we still using SearchContainer?

File: met-web/src/routes/AuthenticatedRoutes.tsx [hunk 11]
```diff
@@ -3,17 +3,12 @@ import { Route, Routes } from 'react-router-dom';
...
 import { ThemeProvider } from '@mui/system';
-import { BaseTheme, PublicTheme } from '../styles/Theme';
+import { BaseTheme} from '../styles/Theme';
 import UserService from '../services/UserServices';
 import View from '../components/Form/View';
-import CreateEngagementForm from '../components/engagement/CreateEngagementForm';
 import Engagement from '../components/engagement';
 
 const AuthenticatedRoutes = () => {
-    let adminRole = UserService.hasAdminRole();
```
Comment (jadmsaadaot) [inline met-web/src/routes/AuthenticatedRoutes.tsx:13]: I understand this creates a lint issue and it is annoying. But maybe it is useful to us to know the future use here? Is it okay to add it again? @saravanpa-aot  thoughts here? Idea is theme={adminRole ? BaseTheme : PublicTheme}.

File: met-web/src/components/layout/SideNav/SideNav.tsx [hunk 12]
```diff
@@ -0,0 +1,70 @@
+import React, { useState } from 'react';
+import { ListItemButton, List, ListItem, ListItemText, Box, Drawer } from '@mui/material';
+import { useNavigate } from 'react-router-dom';
+import { Routes } from './SideNavElements';
+
+const DrawerBox = ({ navigate }) => {
+    const [activeLink, setActiveLink] = useState('/');
+
+    function navigation(path) {
```
Comment (jadmsaadaot) [inline met-web/src/components/layout/SideNav/SideNav.tsx:9]: let's consistently use arrow functions, const navigation = () => {}

File: met-web/src/components/layout/SideNav/SideNav.tsx [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+    return (
+        <Box sx={{ height: '100%', width: '100%', background: '#003366' }} role="presentation">
+            <List sx={{ top: '10%' }}>
+                {Routes.map((route) => (
+                    <ListItem sx={{ height: '50%' }} key={route.name}>
+                        <ListItemButton onClick={() => navigation(route.path)}>
+                            <ListItemText
+                                primaryTypographyProps={{
+                                    variant: 'h6',
+                                    sx: { fontWeight: 'bold', color: activeLink == route.path ? '#ffc107' : '#fafafa' },
```
Comment (jadmsaadaot) [inline met-web/src/components/layout/SideNav/SideNav.tsx:23]: === instead of ==

File: met-web/src/components/layout/Header/index.tsx [hunk 14]
```diff
@@ -1,66 +1,74 @@
...
+    const matches = useMediaQuery((theme) => theme.breakpoints.up('md'));
```
Comment (jadmsaadaot) [inline met-web/src/components/layout/Header/index.tsx:14]: not blocking, but we could improve naming here to isMediumScreenSize

File: met-web/src/routes/LandingPage/LandingPage.tsx [hunk 15]
```diff
@@ -2,16 +2,16 @@ import React from 'react';
...
 import { Link } from 'react-router-dom';
+import TextField from '@mui/material/TextField';
 
 const LandingPage = () => {
     return (
         <Container sx={{ paddingTop: '5em' }}>
             <Grid direction="row" justifyContent="center" alignItems="center" container spacing={2}>
                 <Grid item xs={9}>
-                    <SearchContainer></SearchContainer>
+                    <TextField sx={{ fontSize: '40px' }} placeholder={'Search Engagements by name...'} />
```
Comment (jadmsaadaot) [inline met-web/src/routes/LandingPage/LandingPage.tsx:14]: I still see us tapping into the fontSize prop here, which we should avoid

File: met-web/src/components/layout/Header/index.tsx (hunk 9 above)
Comment (djnunez-aot) [reply met-web/src/components/layout/Header/index.tsx]: Open is needed for smaller screens width has been removed

Comment (jadmsaadaot) [reply met-web/src/components/layout/Header/index.tsx]: of course, I was referring to line 15 the width, setWidth

**Comment by jadmsaadaot** [comment]
Replaced by https://github.com/bcgov/met-public/pull/166 which has conflicts resolved.

**Pull request closed without merging**